mtp_api/apps/notification/management/commands/send_notification_report.py

```diff
@@ -0,0 +1,297 @@
+                count += 1
+        return count
+
+
+def create_zip(period_filename, bundle_path):
+    report_paths = bundle_path.glob('*.csv')
+    report_paths = filter(lambda p: p.stat().st_size > 0, report_paths)
+    report_paths = sorted(report_paths)
+    if not report_paths:
+        return None
+
+    zip_path = bundle_path / f'{period_filename}.zip'
+    with zipfile.ZipFile(zip_path, 'w') as z:
+        for path in report_paths:
+            z.write(path, path.name)
+    return zip_path
+
+
+def send_zip(period_description, zip_file, emails):
+    email = AnymailMessage(
+        subject=f'Prisoner money notifications for {period_description}',
+        body=f'''
+OFFICIAL SENSITIVE
+
+Please find attached, the prisoner money notifications reports for {period_description}.
+
+If you have any queries, contact {settings.TEAM_EMAIL}.
+        '''.strip(),
+        from_email=default_from_address(),
+        to=emails,
+        tags=['notifications-report'],
+    )
+    email.attach_file(str(zip_file), mimetype='application/zip')
+    email.send()
+
+
+class Serialiser:
+    def __init__(self, rule):
+        self.rule = rule
+        self.monitored_rule = isinstance(rule, MonitoredRule)
+
+    def get_headers(self):
+        headers = [
+            'Notification rule',
+            'Internal ID',
+        ]
+        if self.monitored_rule:
+            headers.insert(1, 'Monitored by')
+        return headers
+
+    def serialise(self, record):
+        row = [
+            self.rule.description.replace('you are monitoring', 'someone monitors'),
+            self.internal_id(record),
+        ]
+        if self.monitored_rule:
+            row.insert(1, self.rule.get_event_trigger(record).get_monitoring_users().count())
+        return row
+
+    def internal_id(self, record):
+        raise NotImplementedError
+
+
+class CreditSerialiser(Serialiser):
+    def get_headers(self):
+        return super().get_headers() + [
+            'Date received', 'Date credited',
+            'Amount',
+            'Prisoner number', 'Prisoner name', 'Prison',
+            'Sender name', 'Payment method',
+            'Bank transfer sort code', 'Bank transfer account', 'Bank transfer roll number',
+            'Debit card number', 'Debit card expiry', 'Debit card billing address',
+            'Email', 'IP address',
+            'Status',
+            'NOMIS transaction',
+        ]
+
+    def serialise(self, record: Credit):
+        credited_at = find_log_date(record, CREDIT_LOG_ACTIONS.CREDITED)
+        return super().serialise(record) + [
+            format_csv_datetime(record.received_at), format_csv_datetime(credited_at),
+            format_amount(record.amount),
+            record.prisoner_number, record.prisoner_name, record.prison.short_name,
+        ] + self.serialise_sender(record) + [
+            CREDIT_STATUS.for_value(record.status).display,
+            record.nomis_transaction_id,
+        ]
```

Comment:
I usually tend to use a [DictWriter](https://docs.python.org/3/library/csv.html#csv.DictWriter) to avoid the extra complexity of remembering which header goes with which row.
The serialise could create a dict per row and then you can get the first record, get its keys and you have the headers for free.

It's also easier to read instead of having to figure out what the value '' relates to.